templates/common/_base/units/machine-config-daemon-firstboot-v42.service

```diff
@@ -0,0 +1,21 @@
+name: "machine-config-daemon-firstboot-v42.service"
+enabled: true
+contents: |
+  [Unit]
+  Description=Machine Config Daemon Firstboot (4.2 bootimage)
+  # Make sure it runs only on OSTree booted system
+  ConditionPathExists=/run/ostree-booted
+  BindsTo=ignition-firstboot-complete.service
+  ConditionPathExists=/etc/ignition-machine-config-encapsulated.json
+  # Note the opposite of this in machine-config-daemon-firstboot
+  ConditionPathExists=!/sysroot/.coreos-aleph-version.json
+  After=ignition-firstboot-complete.service
+  Before=kubelet.service
+
+  [Service]
+  # Need oneshot to delay kubelet
+  Type=oneshot
+  ExecStart=/usr/libexec/machine-config-daemon pivot
```

Comment:
this may be dumb, but will this run in 4.1 as well on this conditionpathexists? 4.1 has pivot service in the OS directly so that's gonna be run but this will as well or am I making this up?

Reply:
That's a good question.  So...I think what's going on here right now is you'll notice that
`machine-config-daemon-host.service` does:
```
  # If pivot exists, defer to it.  Note similar code in update.go
  ConditionPathExists=!/usr/lib/systemd/system/pivot.service
```

I think the idea there was that this `firstboot.service` will try to do the "other stuff" like kernel arguments and the defer the actual "os update' to `pivot.service`.

But I'm honestly not sure that really makes sense - we might as well take over the whole thing?

This needs investigation - but we're also doing the same thing that the *current* code is doing so we can't (shouldn't) be making anything worse, right?

Reply:
Yep agreed I think it makes sense and it’ll be skipped indeed 🤔